java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/MemoryUtil.java

```diff
@@ -104,10 +103,7 @@ public Object run() {
                   constructor.setAccessible(true);
                   logger.debug("Constructor for direct buffer found and made accessible");
                   return constructor;
-                } catch (NoSuchMethodException e) {
-                  logger.debug("Cannot get constructor for direct buffer allocation", e);
-                  return e;
-                } catch (SecurityException e) {
```

Comment:
Can you added this back to see if it is okay, I remember this was removed due to netty

Reply:
If we add it back it leads to a `java.lang.reflect.InaccessibleObjectException` -  which was also caused when this change was made - [PR](https://github.com/sirensolutions/arrow/pull/17)

Reply:
did you see this in a test or it is from the linked pr?

Reply:
Yeah, I installed the arrow version locally after making the changes and ran the `InnerJoinTest` on Federate to verify it.